docs/overview.rst

```diff
@@ -591,6 +591,51 @@ example, an interactive mode would be enabled in this run::
       $ TMT_PLUGIN_DISCOVER_FMF_VERBOSE=2 tmt run -a discover -h fmf ...
 
 
+.. _regular-expressions:
+
+Regular expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Many specification keys and command line options accept regular
+expressions, e.g. to filter a set of tests, ``tmt run ... test
+--name '^/foo'``. Whenever tmt works with regular expressions, the
+following rules apply.
+
+Python implementation
+    Since tmt is implemented in Python, Python's `re`__ package is
+    used for handling regular expressions. It comes with Python
+    standard library, and it is widely used and well documented.
+    For the deep dive into supported syntax, see
+    `Regular Expression Syntax`__ section, for a gentler
+    introduction there is a `Regular Expression`__ HOWTO.
+
+Regular expression must match at the beginning of the string
```

Comment:
The heading should probably be updated as well to match the search default. What about this?

```suggestion
Search versus match
```